File: medplum-java/medplum-model/src/main/java/com/fhir/MedicinalProductPharmaceutical.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableMedicinalProductPharmaceutical.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface MedicinalProductPharmaceutical {
  Optional<uri> implicitRules();

  Optional<Meta> meta();

  List<MedicinalProductPharmaceutical_RouteOfAdministration> routeOfAdministration();

  Optional<List<ResourceList>> contained();

  Optional<List<Extension>> extension();

  Optional<List<Reference>> ingredient();

  Optional<code> language();

  Optional<List<Extension>> modifierExtension();

  Optional<List<Reference>> device();

  Optional<Narrative> text();

  String resourceType();

  Optional<id> id();

  Optional<List<Identifier>> identifier();

  CodeableConcept administrableDoseForm();

  Optional<CodeableConcept> unitOfPresentation();

  Optional<List<MedicinalProductPharmaceutical_Characteristics>> characteristics();

  static ImmutableMedicinalProductPharmaceutical.ResourceTypeBuildStage builder() {
    return ImmutableMedicinalProductPharmaceutical.builder();
  }
}

File: medplum-typescript/src/model/fhir/MedicinalProductPharmaceutical.ts
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { MedicinalProductPharmaceutical_Characteristics } from "./MedicinalProductPharmaceutical_Characteristics";
import { MedicinalProductPharmaceutical_RouteOfAdministration } from "./MedicinalProductPharmaceutical_RouteOfAdministration";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { Reference } from "./Reference";
import { ResourceList } from "./ResourceList";
import { uri } from "./uri";

export interface MedicinalProductPharmaceutical {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    identifier: Identifier[] | null | undefined;
    administrableDoseForm: CodeableConcept;
    unitOfPresentation: CodeableConcept | null | undefined;
    ingredient: Reference[] | null | undefined;
    device: Reference[] | null | undefined;
    characteristics: MedicinalProductPharmaceutical_Characteristics[] | null | undefined;
    routeOfAdministration: MedicinalProductPharmaceutical_RouteOfAdministration[];
}
